protocols/dcutr/examples/client.rs

```diff
@@ -186,69 +186,75 @@ fn main() -> Result<(), Box<dyn Error>> {
+    // our local public address and (b) enable a freshly started relay to learn its public address.
+    swarm.dial(opts.relay_address.clone()).unwrap();
     block_on(async {
         let mut learned_observed_addr = false;
-        let mut relay_accepted_reservation = false;
+        let mut told_relay_observed_addr = false;
 
         loop {
             match swarm.next().await.unwrap() {
                 SwarmEvent::NewListenAddr { .. } => {}
                 SwarmEvent::Dialing { .. } => {}
                 SwarmEvent::ConnectionEstablished { .. } => {}
                 SwarmEvent::Behaviour(Event::Ping(_)) => {}
-                SwarmEvent::Behaviour(Event::Relay(client::Event::ReservationReqAccepted {
-                    ..
-                })) => {
-                    info!("Relay accepted our reservation request.");
-                    relay_accepted_reservation = true
+                SwarmEvent::Behaviour(Event::Identify(IdentifyEvent::Sent { .. })) => {
+                    info!("Told relay its public address.");
+                    told_relay_observed_addr = true;
                 }
-                SwarmEvent::Behaviour(Event::Identify(IdentifyEvent::Sent { .. })) => {}
                 SwarmEvent::Behaviour(Event::Identify(IdentifyEvent::Received {
                     info: IdentifyInfo { observed_addr, .. },
                     ..
                 })) => {
-                    info!("Relay observes us under the address: {:?}", observed_addr);
+                    info!("Relay told us our public address: {:?}", observed_addr);
                     learned_observed_addr = true;
                 }
                 event => panic!("{:?}", event),
             }
 
-            // Check whether we are done.
-
-            if !learned_observed_addr {
-                continue;
-            }
-
-            if opts.mode == Mode::Listen && !relay_accepted_reservation {
-                continue;
+            if learned_observed_addr && told_relay_observed_addr {
+                break;
             }
-
-            break;
         }
     });
 
-    if opts.mode == Mode::Dial {
-        swarm
-            .dial(
-                opts.relay_address
-                    .clone()
-                    .with(Protocol::P2pCircuit)
-                    .with(Protocol::P2p(opts.remote_peer_id.unwrap().into())),
-            )
-            .unwrap();
+    match opts.mode {
+        Mode::Dial => {
+            swarm
+                .dial(
+                    opts.relay_address
+                        .clone()
+                        .with(Protocol::P2pCircuit)
+                        .with(Protocol::P2p(opts.remote_peer_id.unwrap().into())),
+                )
+                .unwrap();
+        }
+        Mode::Listen => {
+            swarm
+                .listen_on(opts.relay_address.clone().with(Protocol::P2pCircuit))
+                .unwrap();
```

Comment:
```suggestion
                    opts.relay_address
                        .with(Protocol::P2pCircuit)
                        .with(Protocol::P2p(opts.remote_peer_id.unwrap().into())),
                )
                .unwrap();
        }
        Mode::Listen => {
            swarm
                .listen_on(opts.relay_address.with(Protocol::P2pCircuit))
                .unwrap();
```
Nit: don't think we need to clone the address here; it's not used afterwards anymore.